### ISSUE TITLE
Usability: allow user to try GPS button multiple times

### ISSUE THREAD

**Issue by fancyham** [hackforla/food-oasis#1205]
### Overview

It seems like if I deny access to my location when first loading the site, the site still thinks that I have denied it permanently and the GPS button doesn’t work for the rest of the session/ The button looks clickable (blue background) but it doesn’t seem to do anything.

### To recreate:

Mac Safari (Monterey, latest version of Safari)
1. Click ‘Use my current location’, but do NOT select ‘remember for a day"
2. When Safari asks, deny access to location
3. Type in ‘LA’ and get map
4. GPS button looks active. Click it.
5. Nothing happens. 

Expected result”
That because the GPS button was active, that the website would request my location again and I would see a Safari location request dialog again.
If the button is non-functional, we should make it look disabled.

### Suggested solution 
Try again to get the location whenever the user requests it and show a message when the browser refuses to get or request it.

Or disable the GPS button once user first rejects GPS, but if user reloads site, forget the previous rejection and allow trying to get GPS location again. 

Why:
I may have rejected the GPS but changed my mind.

### Action Items

- [x] Consider suggested solution
- [x] Suggest and discuss alternatives, if any
- [x] Implement change
-

**Comment by staceyrebekahscott** [comment]
Moved to Bugs Column, as the milestone identified is Bug Fixes.

**Comment by andyarensman** [comment]
@fancyham I can fix the issue of the GPS button being clickable when a user has not allowed their location to be used. However, we can not re-request access to a user's location once they have denied permission. The user would need to manually go into their browser settings to enable our site to access their location. The way to do this depends on what browser they are using.

Having the user figure this out on their own is probably not the best user experience, so I'm wondering what the design team thinks would be the best solution to this problem. In the browser version of Google Maps, they have a tool tip that links to a Q&A page for solving the problem:

![rLAgqCkTuj](https://user-images.githubusercontent.com/86622025/225139336-e4162c70-9547-4612-aa80-0ca4514b94f6.jpg)

Clicking on the 'lean more' link brings the user to the following help page with how to update their permissions: https://support.google.com/maps/answer/2839911?hl=en&authuser=0&visit_id=21678828803079-9027348763302606076&co=GENIE.Platform%3DDesktop&oco=&p=browser_lp&rd=1#permission

Currently our tooltip for the disabled location button just says, "Please allow location access." So we could try one of the following:
- Have our tooltip link to our Q&A page and add information on how to give location permissions
- Link to the google maps Q&A above as a temporary fix, though this might confuse users
- Come up with another solution

Please let me know your thoughts and if I can explain anything more clearly.

**Comment by fancyham** [comment]
On iOS 16.4 (beta) Safari, it seems to allow me to deny location access on the first GPS access, then I can reload the page, try again and it'll ask me again (rather than ignoring the request)… but perhaps it's different on other devices/browsers?

Here's an example — it doesn't always work, though, so not sure what the criteria Apple is using to allow/deny:

1. I load foodoasis.la
2. I press the 'locate me' button
3. I get a dialog asking for location permission.  I deny access to my location
4. 'locate me' button is disabled
5. I reload the page
6. I press the 'locate me' button
7. I get a dialog asking for location permission, again.  This time I allow access to my location and it works

https://user-images.githubusercontent.com/3376957/225801512-ae4f571c-1b1c-4f7b-b093-5b33056663f1.MP4

Also, I noticed that if I deny access to my location the first time in step 3, I don't get a message indicating that the web browser doesn't seem to have access to the location…

So is it possible to have some indicator/message explaining that we don't have access to user's location when they try pressing the button again, if we've been denied access?

**Comment by andyarensman** [comment]
The GPS access permissions are different depending on what browser/device you are using and what settings you are using. For example, on Chrome when a user denies access to their location, our app will not be able to ask for their location again unless they go into Chrome settings and allow it. On your device, it looks like it is tied to the session so when you refresh it forgets that you denied permission earlier. So different users will have different experiences.

On the current build of Food Oasis, the GPS button isn't working correctly when the user denies location access. Once I commit the changes I have made, the GPS button will be greyed out/disabled after location access is denied. When the user hovers over the button they will get the message "Please allow location access" as seen below:

![GTlOqjcAYn](https://user-images.githubusercontent.com/86622025/225985627-9ae8510b-51ec-48c0-a04b-132247766d77.jpg)

So my question is, is this enough information for the user, or does Food Oasis need a guide to tell them how to turn the location permissions back on based on what device/browser they are using? If a guide is needed, that should probably be a new issue card for the writers/designers I would think.

**Comment by fancyham** [comment]
So if it’s possible, (and I doubt it is) my preference would be:

1. Link to the OS/browser’s preference screen that lets user enable location access right there (probably impossible)
2. A link to the proper instructions that are appropriate for that browser (but that’s a lot of work, logic, and maintenance, so I think not realistic for us)
3. Perhaps to an item in the FAQ that has those links. (but again, a lot of work keeping that up-to-date)
4. The message you have now is pretty good

Realistically, the user is likely to just use their fingers to navigate to their desired location, so they have an easy workaround, but previously there was no explanation for why the button wasn’t working, so what you have is a big improvement.

So TL;DR: I’m thinking # 4 is what we go with, yes? :)

**Comment by andyarensman** [comment]
# 4 works for me. # 1 is not possible like you thought, and I agree with your points on 2 and 3.

Thanks, Bryan
